src/librustc_typeck/astconv.rs

```diff
@@ -702,6 +702,7 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
                             Some(arg.id()),
                             arg.span(),
                             false,
+                            |_, _| (),
```

Comment:
Is there a good reason that generic arguments have to be special cased in this way?

Reply:
I'm not 100% sure. It avoids a parameter being unmarked, I suspect the reason it's unmarked is https://github.com/rust-lang/rust/pull/65083#issuecomment-573485214. I'm not aware of a work around for it. Any ideas?

Reply:
I'm not quite sure I understood what you meant by that comment. `lookup_stability` will look up the `def_id` of the generic parameter. This should usually be in the crate in which the stability attribute is defined. Do you think that when `Self` is used, the generic parameter actually points to the implicit `Self` type parameter attached to the trait, which is in the user's crate, rather than the crate of definition of the trait itself? If this seems plausible, could we test it? I'm not comfortable special-casing something without knowing why it has to be done that way (and leaving a comment for anyone coming afterwards). Thank you!

Reply:
My thinking is along those lines, it's referring to `T` in `impl<T>`, `fn foo<T>` and the like, not the source crate.

On the previous broken version of this pr I tested this extensively, I should still have the build logs somewhere, I try to review them and possible modernize them.